doc/example/basic.txt

```diff
@@ -237,47 +237,14 @@ If you wish to disable this feature, you can pass the command line option
 Integration with setuptools/distribute test commands
 ----------------------------------------------------
 
-Distribute/Setuptools support test requirements
-and you can extend its test command to trigger
-a test run when ``python setup.py test`` is issued::
-
-    from setuptools.command.test import test as TestCommand
-    import sys
-
-    class Tox(TestCommand):
-        user_options = [('tox-args=', 'a', "Arguments to pass to tox")]
-        def initialize_options(self):
-            TestCommand.initialize_options(self)
-            self.tox_args = None
-        def finalize_options(self):
-            TestCommand.finalize_options(self)
-            self.test_args = []
-            self.test_suite = True
-        def run_tests(self):
-            #import here, cause outside the eggs aren't loaded
-            import tox
-            import shlex
-            args = self.tox_args
-            if args:
-                args = shlex.split(self.tox_args)
-            tox.cmdline(args=args)
-
-    setup(
-        #...,
-        tests_require=['tox'],
-        cmdclass = {'test': Tox},
-        )
-
-Now if you run::
-
-    python setup.py test
-
-this will install tox and then run tox. You can pass arguments to ``tox``
-using the ``--tox-args`` or ``-a`` command-line options. For example::
-
-    python setup.py test -a "-epy27"
-
-is equivalent to running ``tox -epy27``.
+.. warning::
+
+  Integrating tox into ``setup.py`` is strongly discouraged as 
+  it breaks standard packaging approaches as used by many downstream distributions
+  which expect ``setup.py test`` to to test the current interpreter
```

Comment:
@bittner I agree it sounds biased and should therefore be rephrased. I don't think anyone has a good overview of how downstream distributions deal with this or similar situations. Perhaps we should document this somewhere? But where? And who will keep this up to date, as such information will always lag on the current practices of the distributions. 

[PEP 426](https://www.python.org/dev/peps/pep-0426) mentions the following:
```
python setup.py test : run the distribution's test suite in place given an sdist, source archive or VCS checkout
```
The distribution means here the package.

Neither this PEP, nor others (as far as I know), mention what the test suite can and cannot cover. Therefore, using `tox` like that is supposedly fine.

However, as a Python on Nix packager that would very much annoy me and I would argue that
>  `python setup.py test` is for testing against the current env, and `tox` is for testing against
multiple envs.

https://github.com/tox-dev/tox/issues/330#issuecomment-251644538